instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/metrics/db/DbConnectionPoolMetrics.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.metrics.db;
+
+import static io.opentelemetry.api.common.AttributeKey.stringKey;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleHistogram;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.MeterBuilder;
+import io.opentelemetry.api.metrics.ObservableLongUpDownCounter;
+import io.opentelemetry.instrumentation.api.internal.EmbeddedInstrumentationProperties;
+import java.util.function.LongSupplier;
+
+/**
+ * A helper class that models the <a
+ * href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/database-metrics.md#connection-pools">database
+ * client connection pool metrics semantic conventions</a>.
+ */
+public final class DbConnectionPoolMetrics {
+
+  static final AttributeKey<String> POOL_NAME = stringKey("pool.name");
+  static final AttributeKey<String> CONNECTION_STATE = stringKey("state");
+
+  static final class ConnectionState {
```

Comment:
This is a weird enum. I would just move them up to the outer class and prefix with `CONNECTION_`.

Reply:
At first I wanted to imitate how the `SemanticAttributes` look, but later I made these package-private, so it does not matter anymore. Will change 👍